src/ceph-disk/ceph_disk/main.py

```diff
@@ -1541,63 +1541,75 @@ def zap(dev):
     if not stat.S_ISBLK(dmode) or is_partition(dev):
         raise Error('not full block device; cannot zap', dev)
     try:
-        # Thoroughly wipe all partitions of any traces of
-        # Filesystems or OSD Journals
-        #
-        # In addition we need to write 10M of data to each partition
-        # to make sure that after re-creating the same partition
-        # there is no trace left of any previous Filesystem or OSD
-        # Journal
+        if not FREEBSD:
```

Comment:
could you instead separate this big block in a separate function called zap_linux ? and have another function zap_feebsd. It would be easier to maintain don't you think ?